Softeq.XToolkit.WhiteLabel.Droid/Providers/ContextProvider.cs

```diff
@@ -1,22 +1,48 @@
 // Developed by Softeq Development Corporation
 // http://www.softeq.com
 
+using System;
 using Android.App;
 using Android.Content;
+using Softeq.XToolkit.Common.Weak;
+
+#nullable disable
 
 namespace Softeq.XToolkit.WhiteLabel.Droid.Providers
 {
     public interface IContextProvider
     {
+        event EventHandler CurrentActivityChanged;
+
         Activity CurrentActivity { get; }
 
         Context AppContext { get; }
     }
 
     public class EssentialsContextProvider : IContextProvider
     {
+        private WeakReferenceEx<Activity> _activityRef;
+
+        public EssentialsContextProvider()
+        {
+            _activityRef = WeakReferenceEx.Create(CurrentActivity);
+            Xamarin.Essentials.Platform.ActivityStateChanged += OnPlatformActivityStateChanged;
```

Comment:
I'd say we could also expose original `ActivityStateChanged` event here, and let the user of the API decide how to use it.. 
BTW, this way we could implement method `WaitForCurrentActivityChangeAsync()` as an extension to IContextProvider interface